slim/export_inference_graph.py

```diff
@@ -0,0 +1,122 @@
+def with the variables inlined as constants using:
+
+bazel build tensorflow/python/tools:freeze_graph
+bazel-bin/tensorflow/python/tools/freeze_graph \
+--input_graph=/tmp/inception_v3_inf_graph.pb \
+--input_checkpoint=/tmp/checkpoints/inception_v3.ckpt \
+--input_binary=true --output_graph=/tmp/frozen_inception_v3.pb \
+--output_node_names=InceptionV3/Predictions/Reshape_1
+
+The output node names will vary depending on the model, but you can inspect and
+estimate them using the summarize_graph tool:
+
+bazel build tensorflow/tools/graph_transforms:summarize_graph
+bazel-bin/tensorflow/tools/graph_transforms/summarize_graph \
+--in_graph=/tmp/inception_v3_inf_graph.pb
+
+To run the resulting graph in C++, you can look at the label_image sample code:
+
+bazel build tensorflow/examples/label_image:label_image
+bazel-bin/tensorflow/examples/label_image/label_image \
+--image=${HOME}/Pictures/flowers.jpg \
+--input_layer=input \
+--output_layer=InceptionV3/Predictions/Reshape_1 \
+--graph=/tmp/frozen_inception_v3.pb \
+--labels=/tmp/imagenet_slim_labels.txt \
+--input_mean=0 \
+--input_std=255 \
+--logtostderr
+
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+
+from tensorflow.python.platform import gfile
+from datasets import dataset_factory
+from nets import nets_factory
+
+
+slim = tf.contrib.slim
+
+tf.app.flags.DEFINE_string(
+    'model_name', 'inception_v3', 'The name of the architecture to save.')
+
+tf.app.flags.DEFINE_boolean(
+    'is_training', False,
+    'Whether to save out a training-focused version of the model.')
+
+tf.app.flags.DEFINE_integer(
+    'default_image_size', 224,
+    'The image size to use if the model does not define it.')
+
+tf.app.flags.DEFINE_string('dataset_name', 'imagenet',
+                           'The name of the dataset to use with the model.')
+
+tf.app.flags.DEFINE_integer(
+    'labels_offset', 0,
+    'An offset for the labels in the dataset. This flag is primarily used to '
+    'evaluate the VGG and ResNet architectures which do not use a background '
+    'class for the ImageNet dataset.')
+
+tf.app.flags.DEFINE_string(
+    'output_file', '', 'Where to save the resulting file to.')
+
+tf.app.flags.DEFINE_string(
+    'dataset_dir', '', 'Directory to save intermediate dataset files to')
+
+FLAGS = tf.app.flags.FLAGS
+
+
+def main(_):
+  if not FLAGS.output_file:
+    raise ValueError('You must supply the path to save to with --output_file')
+  tf.logging.set_verbosity(tf.logging.INFO)
+  with tf.Graph().as_default() as graph:
+    dataset = dataset_factory.get_dataset(FLAGS.dataset_name, 'validation',
+                                          FLAGS.dataset_dir)
+    network_fn = nets_factory.get_network_fn(
+        FLAGS.model_name,
+        num_classes=(dataset.num_classes - FLAGS.labels_offset),
+        is_training=FLAGS.is_training)
```

Comment:
MobileNet needs another parameter [`depth_multiplier`](https://github.com/tensorflow/models/blob/5eab06018c51d0ec8bf85919bf50438e2c4056a3/slim/nets/mobilenet_v1.py#L269) to build all the variants.